app/models/course/assessment/submission.rb

```diff
@@ -92,6 +92,25 @@ def auto_grade!
     AutoGradingJob.perform_later(self)
   end
 
+  # Return the next unanswered question.
+  #
+  # @return [Course::Assessment::Question|nil] the next unanswered question or nil if all
+  #   questions have been correctly answered.
+  def next_unanswered_question
+    all_questions = assessment.questions
+    correctly_answered_questions =
+      all_questions.where(id: answers.where(correct: true).select(:question_id))
```

Comment:
So that I will use `assessment.questions.next_unanswered(submission)` to get the unanswered question, and code like `correctly_answered_questions` can be reused.